### PR TITLE
New Embed modal

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes https://github.com/metabase/metabase/issues/50552

### Description

This PR implements the new embed modal following [these scenarios.](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c90180b8bce5fca68e6d5ee6)

### How to verify

Follow [these scenarios](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c90180b8bce5fca68e6d5ee6). Each scenario already explains what's the expectation is.

### Demo

image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx:102]: [new test](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c90180748578f93e02f89579): feature unavailable

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx:132]: [new test](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c901806db045fe99a5aae7d9): disabled

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx:157]: [new test](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c901805d80e2ef266b6dbdff): enabled

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx]: Replaced by [2 new SDK tests](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c901803a86c9d0528cf92733)

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.enterprise.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.enterprise.unit.spec.tsx:13]: Test interactive embedding the same way we test it in the `common` test file. This is to ensure the feature isn't leaked to EE build without a valid token (cloud starter)

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.premium.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.premium.unit.spec.tsx:19]: [new test](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c9018075b200f95110f8f407): disabled

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.premium.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.premium.unit.spec.tsx:46]: [new test](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c90180bd8559cdd4cf7acfac): enabled

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx:23]: these 3 new properties except `props` are new.

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx:61]: The change here is to use `mockSetting` and set the value in the `storeInitialState` because our test relies on 2 things.

`MetabaseSetting.get` is achieved by calling `mockSettings`, and if our code read the token features from the state, we need to also mock these token features in the store.

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/setup.tsx:68]: Need to wrap this in a router to test links

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:113]: This is to avoid making the whole thing clickable when the setting is disabled. But rather than setting `pointer-events: none` since we need to be able to click the link that will lead us to enable the setting in the admin page at the bottom of the card.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:115]: these 2 props control the disabled state and the link that the bottom link will bring users to. All 3 embedding options are controlled the same way.
Screenshot 2024-11-28 at 9 16 44 PM

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:125]: When the embedding is disabled (interactive, SDK) instead of rendering the link we don't render it since it makes no sense to render `<a>` without `href` when the embedding is disabled.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:129]: this new flag controls whether or not we will render an anchor tag

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:132]: added for testing

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:149]: only for interactive embedding that has an [**unavailable** state](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c90180748578f93e02f89579) (non pro). We want to lead users to learn more about the features so they could make an informed decision whether or not to buy pro. 

There was a talk to use upsell links here, but it doesn't make much sense since users still haven't understood how interactive embedding works yet, so we think it's better to send them to our embedding landing page instead of forcing them to pay.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:204]: This fixes the link [opening in the same tab unexpectedly](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14969354c90180b0872ef52aa27317e5).

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:209]: `aria-hidden` are to make it more accessible and and we could use accessible queries in testing.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx:16]: just a rename to make it make more sense. I'm not sure what _application embedding_ is

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx:54]: the static embedding option is now controlled by its own embedding flag not `enable-embedding` (old one)

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx:88]: I believe this was a mistake since the described name stated that the resource shouldn't be published, otherwise, we just basically ran the same test 2 times. This change make it so that it test different scenario from the test above.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.unit.spec.tsx:168]: [Test this](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14969354c90180b0872ef52aa27317e5)

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.module.css
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.module.css:3]: this file contains CSS changes for this disabled banner
Screenshot 2024-11-28 at 9 16 44 PM

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx:32]: there are some markup + css changes here for the disabled banner.
Screenshot 2024-11-28 at 9 16 44 PM

File: frontend/src/metabase/ui/components/typography/List/List.styled.tsx
Comment (WiNloSt) [inline frontend/src/metabase/ui/components/typography/List/List.styled.tsx:13]: This is to allow overriding bullet point texts when the card is disabled.
image

File: package.json
Comment (WiNloSt) [inline package.json:253]: Update because I need this fix https://github.com/testing-library/eslint-plugin-testing-library/pull/740

File: e2e/test/scenarios/embedding/embedding-smoketests.cy.spec.js
Comment (WiNloSt) [inline e2e/test/scenarios/embedding/embedding-smoketests.cy.spec.js:334]: Implement the assertion disabled in https://github.com/metabase/metabase/pull/50513

File: e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js
Comment (WiNloSt) [inline e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js:197]: [test this case](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c9018075b200f95110f8f407): disabled

File: e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js
Comment (WiNloSt) [inline e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js:233]: modify the assertion for [this case](https://www.notion.so/metabase/Streamline-embed-flow-when-embedding-features-are-off-13869354c90180b4bc02ca29ebf61ad2?pvs=4#14c69354c90180748578f93e02f89579): unavailable.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx:35]: I know it's not your code but could we change these to a string value with `24px` or `1.5rem` (rems preferred for me)? Right now this value and the `pt` strike me as magic numbers

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx:42]:
```suggestion
      w="22rem"
```

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx:68]: do we need that extra space? I'd rather use a padding value or something like that

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx:35]: I like the text value with unit since it seems to be much more clear what it's supposed to mean.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.module.css
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.module.css:16]: I don't know how comfortable I feel about this one. I think it might make more sense to just change the color or override it in some way. Your call, but I feel like this paradigm makes more sense when we're working with hover states and one-off semantic colors rather than `text-primary`

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx:68]: I'm cool either way. We seem to use `{" "}` in many places by the way, so it's not unordinary. I'll check if I use different method for this kind of spacing in a different place in this PR.

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.premium.unit.spec.tsx
Comment (oisincoveney) [inline frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.premium.unit.spec.tsx:59]: i think that's the case in all of the tests 🫠

**Review by oisincoveney — APPROVED** [review]
Just a few suggestions but in general - I don't see anything that's blocking this from being released. Feel free to take them or leave them

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (heypoom) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:129]: I think the name is a bit misleading, I thought it means whether we show or hide this link (akin to `show && <MaybeLink />`). Perhaps `shouldRenderAsLink` or something more explicit might help.

**Review by heypoom — APPROVED** [review]
I tested each permutations in the embed modal and they work as expected 💯 

CleanShot 2567-12-02 at 21 38 28@2x

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx:1]: @heypoom Link fixed! Thank you for reporting. That happened because I rebased and force pushed.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx:129]: I think `shouldRenderLink` also does the trick.

File: frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/tests/EmbedModalContent.common.unit.spec.tsx:48]: This is an old test, it seems the actual function is create in its parent component, so it couldn't be tested and we need to resort to checking this function's call signature instead.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SelectEmbedTypePane.tsx]: Fixed in 7284ae014ec55e8568eb6ced593bdef365da8649

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx]: Fixed in 004d246310f7ee1f924d2b0808400dda12d2f8c5

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx]: Fixed in 004d246310f7ee1f924d2b0808400dda12d2f8c5

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx]: Fixed in 004d246310f7ee1f924d2b0808400dda12d2f8c5

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.module.css
Comment (WiNloSt) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.module.css:16]: It's hard to override the color that I want to override since it's rendered as `children`.

https://github.com/metabase/metabase/blob/004d246310f7ee1f924d2b0808400dda12d2f8c5/frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/SharingPaneButton/SharingPaneButton.tsx#L51

If you have any better solution in mind feel free to suggest. I think this is a good compromise. I try to pass the disabled class directly to each text we want to dull the color, but the markup is really messy. I think doing this in the CSS like this is way more manageable.

I'll leave this as is for this reason.

**Pull request merged**
